### ISSUE TITLE
Prediction subsampling

### ISSUE THREAD

**Issue by ricklisz** [facebookresearch/capi#11]
Hi, I have a question about the number of prediction tokens. In the config, you have set prediction_subsampling to 0.05. I wonder if this hyperparameter is worth tuning? Would there be any benefit in performance if this number is set to something larger, say 0.5 or 1?

**Comment by TimDarcet** [comment]
Hi,
prediction subsampling is really about efficiency: forward less tokens, but your gradients have higher variance.
0.05 was too low. In the ablations iirc we observed that it would be better to have it higher, I'd say 0.2 for example. The gap won't be huge, but I recon it should be positive.

**Comment by TimDarcet** [comment]
Hi,
It's true that the target entropy is pretty high in all your cases. In my trainings, my target entropy usually was stable around 0.5.
I'm not certain we should expect it to go down though: since we are doing a clustering on a moving distribution, there's no guarantee that it will decrease. I think we just want the entropy to be relatively stable.
I'd say first check if the model seems good: do you have downstream tasks you could evaluate on, or can you visually inspect some PCAs?
If the model is not good, we can further try debugging the clustering

**Comment by ricklisz** [comment]
Gotcha. I have tested the model on my downstream (which is a segmentation task) and my observation is that pretraining with the entropy going down did perform slightly better (+0.1 dice score). However, I do suspect the model is not good, as it cannot outperform a supervised baseline (0.4 dice score lower), which is quite strange. As you mentioned the clustering may have issue, what suggestion would you have to debug this step? 

For a bit more context, I am using a per GPU batch size of 20, with 2 GPUs so global batch size is 40. Since our input is 3D volume of H x W x D, 20 is the maximum batch available for a 80GB GPU. **I made no other changes to the hyperparameters other than adjusting the corresponding learning rate**, which I set to 0.0003 for the model and half of that for clustering. 

Here are the evaluations on my segmentation:
Pretraining with entropy goes down: DSC = 0.8769
Pretraining with entropy goes up: DSC = 0.8754
Supervised (without pretraining): DSC = 0.8806

**Comment by TimDarcet** [comment]
Ah, if your batch size is 40 that could be a problem. My batch size was 16384 for all experiments, I recommend you try and obtain a bigger batch size somehow. You could use a smaller model, or a bigger patch size, I think it would make sense.

Clarification: the DSC results after CAPI pretraining, are they using linear probing or full finetuning?
